docs/recipes/nuxt.md

```diff
@@ -0,0 +1,123 @@
+# Testing Vue.js components with Nuxt.js
+
+## Dependencies
+
+- [Require extension hooks](https://github.com/jackmellis/require-extension-hooks):
+	- `npm i --save-dev require-extension-hooks require-extension-hooks-vue require-extension-hooks-babel@beta`
+
+- [browser-env](browser-testing.md)
+	- `npm i --save-dev browser-env`
+
+- [babel-plugin-module-resolver](https://github.com/tleunen/babel-plugin-module-resolver#readme)
+	- `npm i --save-dev babel-plugin-module-resolver`
+
+## Setup
+
+First set up a helper to configure the environment to transpile `.vue` files and run in a browser like environment.
+
+**`package.json`:**
+
+```json
+{
+	"ava": {
+		"require": [
+			"./test/_setup.js"
+		]
+	}
+}
+```
+
+```js
+// ./test/_setup.js
+
+// Setup browser environment
+require('browser-env')();
+const hooks = require('require-extension-hooks');
+const Vue = require('vue');
+
+// Setup Vue.js to remove production tip
+Vue.config.productionTip = false;
+
+// https://github.com/nuxt/create-nuxt-app/issues/180
+window.Date = global.Date = Date;
+
+// Setup `.vue` files to be processed by `require-extension-hooks-vue`
+hooks('vue').plugin('vue').push();
+// Setup `.vue` and `.js` files to be processed by `require-extension-hooks-babel`
+hooks(['vue', 'js']).exclude(({filename}) => filename.match(/\/node_modules\//)).plugin('babel').push();
+```
+
+**`babel.config.js`:**
+```js
+module.exports = {
+  "env": {
+    "test": {
+      "plugins": [
+        [
+          "module-resolver",
+          {
+            "root": [
+              "."
+            ],
+            "alias": {
+              "@": ".",
+              "~": "."
+            }
+          }
+        ]
+      ],
+      "presets": [
+        [ "@babel/env", {
+          targets: {
+            node: "current"
+          }
+        }]
+      ]
+    }
+  },
+};
+```
+
+You can find more information about setting up Babel with AVA in the [Babel recipe](babel.md).
+
+## Sample snapshot test
+
+```js
+import {mount, createLocalVue} from '@vue/test-utils';
+import test from 'ava';
+import Index from '@/pages/index.vue';
+
+let wrapper;
+const localVue = createLocalVue();
+
+test.beforeEach(() => {
+	wrapper = mount(Index, {
```

Comment:
You cannot use a shared variable in a `beforeEach()` hook. It works in this example, because the tests are synchronous. But if you had asynchronous tests then the `beforeEach` hooks would run and change the `wrapper` value.

If it can be shared, use `test.before()`. But perhaps better yet is to assign it to the context:

```js
test.beforeEach(t => {
  t.context.wrapper = mount(Index, { localVue })
})

test('is a Vue instance', t => {
  t.true(t.context.wrapper.isVueInstance())
})
```